fs/tarfs.py

```diff
@@ -204,59 +204,55 @@ def __str__(self):
         return "<TarFS '{}'>".format(self._file)
 
     def getinfo(self, path, namespaces=None):
-        self.check()
+        _path = relpath(self.validatepath(path))
         namespaces = namespaces or ()
-        path = relpath(normpath(path))
-        if not path:
-            raw_info = {
-                "basic":
-                {
-                    "name": "",
-                    "is_dir": True,
-                },
-                "details":
-                {
+        raw_info = {}
+
+        if _path in '/':
```

Comment:
If `_path` has gone through `relpath` then I think `_path` could never be "/".

It should still trip with the empty string, but perhaps `not path` or `path == ""` is more explicit?

Reply:
At start I did only `_path = self.validatepath(path)`, so I had to check that way, but since after that the failing tests remembered me `TarFile.getmember` expected a relative path, I added relpath and forgot to remove it.